### PR TITLE
SageMaker enhancements to allow custom docker image, input channels referring to s3/remote data locations and metrics logging

### PR DESCRIPTION
**Pull request by pacman100**
### What does this PR do?
Addresses the requests of #502 and more
1. One can specify a custom docker image instead of Official 🤗 DLCs through the `accelerate config` questionnaire. When this isn't provided, the latest Official 🤗 DLC will be used.
2. Support for input channels pointing to S3 data locations via TSV file, e.g., below are the contents of `sagemaker_inputs.tsv` whose location is given as part of `accelerate config` setup. 
```tsv
channel_name	data_location
train	s3://sagemaker-sample/samples/datasets/imdb/train
test	s3://sagemaker-sample/samples/datasets/imdb/test
```
3. Support for SageMaker metrics logging via TSV file, e.g., below are the contents of the `sagemaker_metrics_definition.tsv` whose location is given as part of `accelerate config` setup.
```tsv
metric_name	metric_regex
accuracy	'accuracy': ([0-9.]+)
f1	'f1': ([0-9.]+)
```
Below is the logging of the metrics in SageMaker
Screenshot 2022-07-11 at 6 17 25 PM

Example of accelerate config with above features setup [XXXXX values are AWS account specific]:
```bash
base_job_name: accelerate-sagemaker-1
compute_environment: AMAZON_SAGEMAKER
distributed_type: DATA_PARALLEL
ec2_instance_type: ml.p3.16xlarge
iam_role_name: XXXXX
image_uri: 763104351884.dkr.ecr.us-east-1.amazonaws.com/huggingface-pytorch-training:1.8.1-transformers4.10.2-gpu-py36-cu111-ubuntu18.04
mixed_precision: fp16
num_machines: 1
profile: XXXXX
py_version: py38
pytorch_version: 1.10.2
region: us-east-1
sagemaker_inputs_file: /home/ubuntu/sagemaker_examples/sagemaker_inputs.tsv
sagemaker_metrics_file: /home/ubuntu/sagemaker_examples/sagemaker_metrics_definition.tsv
transformers_version: 4.17.0
use_cpu: false
```

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for adding this! LGTM but I'd also like to see @philschmid review since he is the SageMaker expert :-)

File: src/accelerate/commands/config/config_args.py [hunk 1]
```diff
@@ -156,10 +156,13 @@ def __post_init__(self):
 class SageMakerConfig(BaseConfig):
     ec2_instance_type: str
     iam_role_name: str
+    image_uri: str
```
Comment (philschmid) [inline src/accelerate/commands/config/config_args.py:159]: I think we should default to the HF Image?

File: src/accelerate/commands/config/config_args.py [hunk 2]
```diff
@@ -156,10 +156,13 @@ def __post_init__(self):
...
     profile: Optional[str] = None
     region: str = "us-east-1"
     num_machines: int = 1
     base_job_name: str = f"accelerate-sagemaker-{num_machines}"
     pytorch_version: str = SAGEMAKER_PYTORCH_VERSION
     transformers_version: str = SAGEMAKER_TRANSFORMERS_VERSION
     py_version: str = SAGEMAKER_PYTHON_VERSION
+    sagemaker_inputs_file: str = None
+    sagemaker_metrics_file: str = None
```
Comment (philschmid) [inline src/accelerate/commands/config/config_args.py:168]: Does accelerate output some default metrics, e.g. `loss` then we could already define them here? Or is there a "logging" schema, that way we could define a whole bunch of metrics already.

**Review by philschmid — APPROVED** [review]
LGTM ✅

File: src/accelerate/commands/config/config_args.py (hunk 2 above)
Comment (pacman100) [reply src/accelerate/commands/config/config_args.py:168]: `Accelerate` doesn't log any metrics by default, it enables running the same PyTorch code on any setup and as such user is in full control of what is being logged (flexibility).

File: src/accelerate/commands/config/config_args.py (hunk 1 above)
Comment (pacman100) [reply src/accelerate/commands/config/config_args.py:159]: HF image is used by default if `image_uri` isn't specified based on the `Python`, `PyTorch` and `Transformers` versions.

**Comment by plamb-viso** [comment]
Wow @pacman100 you responded to that request quickly. Thanks everyone, the Accelerate team is awesome.

**Pull request merged**